### PR TITLE
Bump develop when release branch created

### LINKED ISSUE #695: Increment "develop" version when branching release
This issue has been broken out of the discussion here: https://github.com/GitTools/GitVersion/issues/632

The current behavior is to use the "-unstable.N" pre-release revision to ensure builds from _develop_ are considered newer than those on release branches that have not yet been merged back to _develop_.

Example:
```
A-B-C (develop)
   \D-E (release/0.1)
```
In the above graph, build A-C would get the versions 0.1.0-unstable.1-3, and D and E would get 0.1.0-rc.1-2. A-C would all be considered *newer* than D and E. Not until _release/0.1_ is merged back into _develop_ does _develop_'s version get bumped to 0.2.0. I find this counter-intuitive. After all, I've branched the release, why is it still being developed (as opposed to just being stabilized), and why are those builds considered newer than the release itself?

One might argue that deleting a release branch, after concluding the release should be cancelled, would then cause _develop_ to drop back(!) to the previous minor version, but I don't think one should ever delete release branches like that. 1) it causes the problem I just mentioned, and 2) if you had at least one build from that release branch before deleting it, and then re-create the same release, there's nothing guarding you from generating the same 0.1.0-rc.1 version again, which would also become confusing. You could of course tag/configure _develop_ to stay at 0.2.0, but that feels like a bit of a kludge, and you still shouldn't re-create releases. There's nothing in SemVer that says you cannot skip a version, AFAIK, so this wouldn't invalidate anything.

If this was to be changed, and _develop_ were to be bumped on branch, then the -unstable tag should be replaced with something that sorts before -rc, but I haven't got a suggestion that is better than the one in https://github.com/GitTools/GitVersion/issues/632.

### PR DESCRIPTION
**Pull request by JakeGinnivan**
- Fixed an commit counting bug
 - Based on discussion at #632
 - Fixes #695
 - Updated related docs

### CODE REVIEW

File: src/GitVersionCore/EffectiveConfiguration.cs [hunk 1]
```diff
@@ -43,8 +44,13 @@ int commitsSinceVersionSourcePadding
             LegacySemVerPadding = legacySemVerPaddding;
             BuildMetaDataPadding = buildMetaDataPadding;
             CommitsSinceVersionSourcePadding = commitsSinceVersionSourcePadding;
+            IsCurrentBranchDevelop = isCurrentBranchDevelop;
+            IsCurrentBranchRelease = isCurrentBranchRelease;
         }
 
+        public bool IsCurrentBranchDevelop { get; private set; }
+        public bool IsCurrentBranchRelease { get; private set; }
```
Comment (gep13) [inline src/GitVersionCore/EffectiveConfiguration.cs]: @JakeGinnivan very small point, but can we get a space between the properties?

**Comment by gep13** [comment]
@JakeGinnivan apart from one small comment, this looks good to me. :+1:

**Comment by asbjornu** [comment]
I think this looks good! Rebase, let Travis and AppVeyor do their thing and if all green; merge?

**Comment by DanielRose** [comment]
When will this be merged? I could really use this.

**Comment by JakeGinnivan** [comment]
This is a bit of a change of behaviour. What does everyone thing, changing defaults ok for a minor release? develop is going from unstable to alpha with this change

**Comment by asbjornu** [comment]
Since people can always configure `develop` to have the `unstable` tag, I think it's fine doing this in a minor. This configuration option should probably be mentioned in the release notes, though, so it's easily accessible information in case it causes trouble for anyone.

**Comment by JakeGinnivan** [comment]
I need to see the impacts. It changes sorting and bumping rules. Changing back to unstable will cause more problems I think.

I think the sorting is the same, or better in most cases once this is merged. Spose we can release as a pre-release and gather feedback.

**Comment by JakeGinnivan** [comment]
@asbjornu have updated some of the images in the docs.

Will do the mainline docs update in that branch

**Comment by JakeGinnivan** [comment]
This shows the real effect of this branch quite well.

image

**Comment by asbjornu** [comment]
@JakeGinnivan That picture makes perfect sense. LGTM! 😄 👍

**Comment by DanielRose** [comment]
@JakeGinnivan One small thing I noticed: In docs/git-branching-strategies/img/39f9d8b8b007c82f1f80_major-release.png (also shown above), the effect of tagging the master branch is not shown anymore.

**Comment by JakeGinnivan** [comment]
Nice catch @DanielRose, R#s test runner truncates output now and I didn't look properly :P

**Comment by JakeGinnivan** [comment]
Have fixed that image in #950. Thanks @DanielRose

**Pull request merged**